src/utils/client.go

```diff
@@ -0,0 +1,264 @@
+	tokenSaltFile := flag.String("salt-file", "/data/secret/keys/uaa_token_salt", "uaa token salt file")
+	uaaServer := flag.String("uaa-server", "", "UAA server to authenticate")
+	verifyCert := flag.Bool("verify-cert", false, "Verify cert")
+	// verbose is used for debugging, dump all information to console
+	verbose := flag.Bool("verbose", false, "Display verbose info")
+
+	flag.Parse()
+
+	trace = *verbose
+
+	if *isShowUAA == true {
+		uaaProf, err := parseJsonFile(*uaaJsonPath)
+		checkError(err)
+		uaaProf.TokenSalt = readTokenSalt(*tokenSaltFile)
+		Trace("token salt is changed to " + uaaProf.TokenSalt)
+		content, err := json.Marshal(uaaProf)
+		fmt.Print(string(content))
+		os.Exit(0)
+	}
+
+	checkParameter(harborServerUrl, password, uaaServer)
+	client := createHttpClient(*trustedCA, *insecure)
+	c := newConfigClient(client, *harborServerUrl, "admin", *password)
+	uaaProf, err := parseJsonFile(*uaaJsonPath)
+	checkError(err)
+	if *isConfigOIDC == true {
+		err = c.configOIDC(uaaProf, *uaaServer, *verifyCert)
+	}
+
+	if *isConfigUAA == true {
+		err = c.configUAA(uaaProf, *uaaServer, *verifyCert)
+
+	}
+	checkError(err)
+	os.Exit(0)
+}
+
+func checkParameter(harborServerUrl, password, uaaServer *string) {
+	if len(*harborServerUrl) == 0 {
+		Trace("Please provide server url to configure")
+		os.Exit(1)
+	}
+	if len(*password) == 0 {
+		Trace("need to provide admin password to configure")
+		os.Exit(1)
+	}
+	if len(*uaaServer) == 0 {
+		Trace("Need to provide the uaa server info")
+		os.Exit(1)
+	}
+}
+
+func checkError(err error) {
+	if err != nil {
+		Trace(err)
+		os.Exit(1)
+	}
+}
+
+func parseJsonFile(jsonPath string) (*UAAProfile, error) {
+	var profile UAAProfile
+	content, err := ioutil.ReadFile(jsonPath)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(content, &profile)
+	if err != nil {
+		return nil, err
+	}
+	return &profile, nil
+}
+
+func readTokenSalt(tokenSaltFile string) string {
+	var token string
+	f, err := os.Open(tokenSaltFile)
+	if os.IsNotExist(err) {
+		token, err = tokenSalt(tokenSaltFile)
+		if err != nil {
+			return ""
+		}
+		return token
+	}
+	defer f.Close()
+	content, _ := ioutil.ReadFile(tokenSaltFile)
+	token = string(content)
+
+	if strings.TrimSpace(token) == "" {
+		token, err = tokenSalt(tokenSaltFile)
+	}
+	return token
+}
+
+func tokenSalt(tokenSaltFile string) (string, error) {
+	token := randString(8)
+	err := ioutil.WriteFile(tokenSaltFile, []byte(token), 0644)
+	if err != nil {
+		return "", nil
+	}
+	return token, nil
+}
+
+func stringWithCharset(length int, charset string) string {
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = charset[seededRand.Intn(len(charset))]
+	}
+	return string(b)
+}
+
+func randString(length int) string {
+	return stringWithCharset(length, charset)
+}
+
+func createHttpClient(caCertFile string, insecure bool) *http.Client {
+	// Get the SystemCertPool, continue with an empty pool on error
+	rootCAs, _ := x509.SystemCertPool()
```

Comment:
if it's insecure, there's no need to inject CA certs.

Reply:
update